### ISSUE TITLE
Invalid DynamoDB state cannot be fixed with new schema.graphql

### ISSUE THREAD

**Issue by yuhui** [aws-amplify/amplify-cli#7772]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-adminui/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-adminui/blob/main/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### App Id

d32bslavnn3g29

### Region

us-east-1

### Environment name

dev

### Amplify CLI Version

5.1.1

### If applicable, what version of Node.js are you using?

16.4.2

### What operating system are you using?

Mac

### Browser type?

Google Chrome

### Describe the bug

I had added a many-to-many relationship in schema.graphql, e.g. A -- AB -- B, where A and B are the main tables and AB is the join table. But this was done wrongly, such that table A did not have the required join field, though table B did. Now, when deploying the app, Amplify returns this error (actual names have been replaced with "A" and "AB"):

> A does not support 1 to 1 connection with both sides of connection as optional field: AB.fieldA

I fixed schema.graphql. But when running `amplify api push`, I still get the same error. That means Amplify is still stuck with the old schema and cannot be recovered.

Deleting the app is also not possible because of this invalid state.

### Expected behavior

When pushing a new schema.graphql, Amplify should apply the new schema and overwrite any previously set models/tables.

### Reproduction steps

1. Push an invalid schema.graphql.
2. Deploy the app and get error message.
3. Fix schema.graphql.
4. Push the updated schema.

**Comment by SwaySway** [comment]
Hello @yuhui 
A few questions here
- Did you create this schema locally or through admin ui?
- Can you send over a reproducible schema? If the information is too sensitive you can send it to amplify-cli@amazon.com. (please include your issue number)

**Comment by yuhui** [comment]
@SwaySway 

> Did you create this schema locally or through admin ui?

I created the schema locally.

> Can you send over a reproducible schema? If the information is too sensitive you can send it to amplify-cli@amazon.com. (please include your issue number)

I have sent this to the email address.

**Comment by josefaidt** [comment]
Hey @yuhui :wave: are you seeing this error on `amplify push`? I was able to push and interact with the provided schema okay, however I can generate a similar error if I explicitly run `amplify codegen models`. Do you have DataStore enabled?

**Comment by yuhui** [comment]
Yes, I had first encountered the error when running `amplify push`. Yes, I had to enable DataStore to work with the models in Admin UI.

Frankly, I'd prefer if I could just delete this bad app once-and-for-all, even if the issue is not resolved. But I can't even delete the app too because of that error. Is there a way to force-delete an Amplify app?

**Comment by josefaidt** [comment]
I was able to reproduce the DataStore error after enabling with the provided schema, as well as the schema from a relevant issue in another repo https://github.com/aws-amplify/amplify-ios/issues/505

I did notice that it was able to push but failed at creating models, and was able to mitigate this by disabling datastore:

1. `amplify update api`
2. Choose "GraphQL"
3. Choose "Disable DataStore for entire API"

However, this appears to be a bug with codegen and DataStore. Using the following schema we see this error surface whether both fields are required or both fields are optional, which was [also mentioned in a similar issue](https://github.com/aws-amplify/amplify-cli/issues/5249#issuecomment-686969050). When one is required and the other is not, this error does not appear.

To reproduce, copy this schema and run `amplify codegen models` or enable DataStore for the entire API

```graphql
type PhoneCall @model {
  id: ID!
  caller: Person! @connection(name: "PhoneCallCaller")
  callee: Person! @connection(name: "PhoneCallCallee")
  transcript: Transcript! @connection(name: "PhoneCallTranscript")
}

type Person @model {
  id: ID!
  name: String!
  callerOf: [PhoneCall!] @connection(name: "PhoneCallCaller")
  calleeOf: [PhoneCall!] @connection(name: "PhoneCallCallee")
}

type Transcript @model {
  id: ID!
  text: String!
  language: String
  phoneCall: PhoneCall! @connection(name: "PhoneCallTranscript")
}
```

**Comment by yuhui** [comment]
Another AWS user has managed to delete the app with this problem, so I was not able to replicate the steps mentioned.

Given that the app has been deleted, for me, this problem is "resolved".

**Comment by josefaidt** [comment]
Hey @yuhui glad to hear you were able to remove the app without further issues 🙌 ! I'll keep this issue open to track the codegen & DataStore -related bug
